tests/integration/basic_data/policy-pack/index.ts

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016-2019, Pulumi Corporation.  All rights reserved.
+
+import * as assert from "assert";
+
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+import { PolicyPack } from "@pulumi/policy";
+
+new PolicyPack("basic-data-policy", {
+    policies: [
+        {
+            name: "basic-data-resource-validation",
+            description: "Verifies data during resource validation.",
+            enforcementLevel: "mandatory",
+            validateResource: (_, __) => {
+                verifyData();
+            },
+        },
+        {
+            name: "basic-data-stack-validation",
+            description: "Verifies data during stack validation.",
+            enforcementLevel: "mandatory",
+            validateStack: (_, __) => {
+                verifyData();
+            },
+        },
+    ],
+});
+
+function verifyData() {
+    // Verify isDryRun().
+    assert.strictEqual(pulumi.runtime.isDryRun(), true, "'isDryRun()' not the expected value");
```

Comment:
Can we be more precise here?

It isn't that `isDryRun()` is _always_ expected to be set. It's that our test harness only performs previews. I think that distinction is important enough to call out with a comment, or maybe even having wiring through another test scenario where we verify these fields during a proper `pulumi up`.